### PR TITLE
Add offline messages suport

### PR DESCRIPTION
**Pull request by postva**

### CODE REVIEW

File: test/chat_test.py [hunk 1]
```diff
@@ -77,6 +85,42 @@ def test_send_chat_message(self):
...
 
+    def test_chat_offline_message(self):
+        print("== Test offline chat messages")
+        firs_offline_message = "first offline message"
+        second_offline_message = "second offline message"
+        self.user4 = self.getLoggedInUser(self.name_pass4)
+        time.sleep(2)
+        print("Send first message")
+
+        self.user4.send_message(
```
Comment (ivenn) [inline test/chat_test.py]: It's better to create messages with constructor from protocol.messages and then call it's method 'as_bytes'

File: server/models/message.py [hunk 2]
```diff
@@ -22,12 +22,16 @@ class ChatMessage(Base):
...
+
     sent_by = Column(Integer, ForeignKey('user.id'))
+    by = relationship('User', foreign_keys=[sent_by, ])
+
 
     def __init__(self, data, chat, sent_by):
         self.data = data
         self.chat = chat
-        self.sent_by = sent_by
+        self.by = sent_by
```
Comment (ivenn) [inline server/models/message.py]: Was it renamed by mistake? self.sent_by -> self.by

Comment (postva) [reply server/models/message.py]: No, I can rename constructor parameter. But original idea was to make ChatMessage class more similar to Message class

Comment (ivenn) [reply server/models/message.py]: I was asking that just because __ repr__ was not changed accordingly.

File: server/models/message.py [hunk 3]
```diff
@@ -27,11 +27,10 @@ class ChatMessage(Base):
...
-
-    def __init__(self, data, chat, sent_by):
+    def __init__(self, data, chat, by):
         self.data = data
         self.chat = chat
-        self.by = sent_by
+        self.by = by
 
     def __repr__(self):
         return "ChatMessage(id={id}, ts={ts}, data={data}, chat={chat}, sent_by={sent_by})".format(
```
Comment (ivenn) [inline server/models/message.py]: Please replace sent_by=self.sent_by with by=self.by

**Review by ivenn — APPROVED** [review]

**Pull request merged**